### ISSUE TITLE
Asymmetric forces for symmetric molecule

### ISSUE THREAD

**Issue by kamikode** [grimme-lab/xtb#174]
For some inputs, the xtb code will produce asymmetric forces even though the molecule is symmetric. I have uploaded an example for an ethene (C2H4) molecule "bad_geometry.xyz". When xtb is run with this input
`xtb --grad bad_geometry.xyz`
the resulting gradient file (also attached) contains asymmetric forces, even though the molecule is symmetric. The force magnitude acting on the H-atoms bound to the first C atom is 0.8282717176e-2, whereas the force magnitude acting on the H-atoms bound to the second C atom is 0.4971962784e-1. The magnitudes should be the same. This is reproducible also with the python API and with different electronic temperature settings.

[example.zip](https://github.com/grimme-lab/xtb/files/4516205/example.zip)

**Comment by awvwgk** [comment]
Not sure if I understand the problem correctly.

It's a D2 symmetric system, so we should find the corresponding symmetry in the gradient. I can spot one due to a rotation axis aligned to a coordinate axis, the other ones are more difficult to carry out by hand in this orientation, therefore I aligned the axes with the coordinate system and everything looks fine to me:

```
$gradient
  cycle =      1    SCF energy =    -6.13841304756   |dE/dxyz| =  0.152669
    1.24378171503439      0.00000000000993     -0.00000000000367      C 
    2.32312325740619      1.26746400709268      1.17339657236287      H 
    2.32312325741949     -1.26746400705560     -1.17339657237660      H 
   -2.32312325739926     -1.26747233543049      1.17338886226769      H 
   -2.32312325742642      1.26747233539341     -1.17338886225397      H 
   -1.24378171503439     -0.00000000000993      0.00000000000367      C 
  -8.1913133442331E-02  -7.8328178221324E-13  -8.2606930773571E-13
  -7.0368882222001E-03  -3.5715846874639E-02   3.3866005356251E-02
  -7.0368882217450E-03   3.5715846874052E-02  -3.3866005356346E-02
   7.0367847849274E-03   3.5715430997772E-02   3.3866385399207E-02
   7.0367847842971E-03  -3.5715430997290E-02  -3.3866385399263E-02
   8.1913340317051E-02   8.8820736789813E-13   9.7790479598563E-13
$end
```

But maybe you can clarify?

**Comment by kamikode** [comment]
Yes, it seems that with your input geometry, the issue is not present.
For your input, the forces acting on the hydrogens all have a magnitude of ~0.4972e-1, see the relevant lines:
`  -7.0368882222001E-03  -3.5715846874639E-02   3.3866005356251E-02
  -7.0368882217450E-03   3.5715846874052E-02  -3.3866005356346E-02
  7.0367847849274E-03   3.5715430997772E-02   3.3866385399207E-02
   7.0367847842971E-03  -3.5715430997290E-02  -3.3866385399263E-02
`
However, in the original orientation I have attached in my original message, the gradient file looks like this:
`$grad                                                                           
  cycle =      1    SCF energy =    -6.13841304696   |dE/dxyz| =  0.136018
   -1.24378171503439      0.00000000000000      0.00000000000000      c 
   -2.32312325741284     -1.72723030464310      0.00000000000000      h 
   -2.32312325741284      1.72723030464310      0.00000000000000      h 
    2.32312325741284      0.13294411025901      1.72210725759455      h 
    2.32312325741284     -0.13294411025901     -1.72210725759455      h 
    1.24378171503439      0.00000000000000      0.00000000000000      c 
   0.8191308748228D-01  -0.2855348065194D-16  -0.4611247364852D-17
   0.7036901509650D-02   0.3201845461387D-02  -0.2972138388538D-02
   0.7036901509650D-02  -0.3201845461387D-02   0.2972138388538D-02
  -0.7036798124096D-02  -0.4921563607436D-01   0.5882435785803D-03
  -0.7036798124096D-02   0.4921563607436D-01  -0.5882435785804D-03
  -0.8191329425339D-01  -0.4297845174368D-17   0.9917739372811D-16
$end                                                                            
`
If you look at the relevant lines
`
   0.7036901509650D-02   0.3201845461387D-02  -0.2972138388538D-02
   0.7036901509650D-02  -0.3201845461387D-02   0.2972138388538D-02
  -0.7036798124096D-02  -0.4921563607436D-01   0.5882435785803D-03
  -0.7036798124096D-02   0.4921563607436D-01  -0.5882435785804D-03
`
and manually calculate the force magnitude by doing sqrt(fx^2+fy^2+fz^2), you will see that the force magnitude on the first two hydrogens is ~0.8283e-2, but the magnitude on the second two hydrogens is ~0.4972e-1, same as your results. So it seems like this bug is only present in the original input. 
Can you calculate the gradient for the original input file and confirm my numbers? It might also help to visualize the forces in 3d, for example using ase gui (this is how I originally spotted the bug).

**Comment by awvwgk** [comment]
> and manually calculate the force magnitude by doing sqrt(fx^2+fy^2+fz^2), you will see that the force magnitude on the first two hydrogens is ~0.8283e-2, but the magnitude on the second two hydrogens is ~0.4972e-1, same as your results. So it seems like this bug is only present in the original input.
Can you calculate the gradient for the original input file and confirm my numbers? It might also help to visualize the forces in 3d, for example using ase gui (this is how I originally spotted the bug).

I see your point, but so far I'm unable to reproduce it on the master branch, with (`--cma`) and without rotating the system the symmetry of the gradient is preserved for me. With the current master I get:

```
$gradient
  cycle =      1    SCF energy =    -6.13841304752   |dE/dxyz| =  0.152669
   -1.24378171503439      0.00000000000000      0.00000000000000      C 
   -2.32312325741284     -1.72723030464310      0.00000000000000      H 
   -2.32312325741284      1.72723030464310      0.00000000000000      H 
    2.32312325741284      0.13294411025901      1.72210725759455      H 
    2.32312325741284     -0.13294411025901     -1.72210725759455      H 
    1.24378171503439      0.00000000000000      0.00000000000000      C 
   8.1912927725238E-02   1.5244051951736E-16   2.1263915107872E-17
   7.0369863671883E-03   3.2019016951934E-03   4.9114922829296E-02
   7.0369863671883E-03  -3.2019016951934E-03  -4.9114922829296E-02
  -7.0366891241343E-03  -4.9215627866832E-02   5.8832572654632E-04
  -7.0366891241344E-03   4.9215627866832E-02  -5.8832572654639E-04
  -8.1913522211346E-02  -7.7927031147396E-19   1.1877434799579E-16
$end
```

Both energy and gradient might differ slightly since I was restarting the calculation from the rotated structure, which certainly did mess up the CAMMs.

Also there has been larger changes with f639aed0c80d4daf93aaff3253575b93f8011291, one was a fix on a small issue with the gradient from the three-body dispersion contribution, which might cause this problem for you.

**Comment by kamikode** [comment]
I tested with the latest pre-compiled binaries, without `--cma` and without restarting from any previous calculation. It seems like the pre-compiled binaries do not have these fixes yet. Thanks!

**Comment by awvwgk** [comment]
Than it might be related to this fix, it will be amended in the pre-compiled version with the release of 6.3.
